### PR TITLE
Sanity checks for Orange/Red range and display values. Makes display reverse and brightness changes real time in preferences.html

### PR DESCRIPTION
**Pull request by Coscolin**
…rse and brightness

### CODE REVIEW

File: CO2_Gadget_WIFI.h [hunk 1]
```diff
@@ -950,6 +950,9 @@ const char *PARAM_INPUT_1 = "MeasurementInterval";
 const char *PARAM_INPUT_2 = "CalibrateCO2";
 const char *PARAM_INPUT_3 = "SetVRef";
 
+// Predeclaration
+void setDisplayBrightness(uint16_t newBrightness);
```
Comment (melkati) [inline CO2_Gadget_WIFI.h:954]: Please, move with all others forward declaration to CO2_Gadget.ino:
https://github.com/melkati/CO2-Gadget/blob/e626693d59fe1f7798dbce906ddfbe1ca08f55ba/CO2_Gadget.ino#L45-L52

File: webserver/preferences.js [hunk 2]
```diff
@@ -595,7 +595,7 @@ document.addEventListener("DOMContentLoaded", () => {
 
     // Check if the current URL contains "preferences.html"
     if (currentURL.includes("preferences.html")) {
-        highlightCurrentPage(); // Highlight the current page in the navigation bar
+        //        highlightCurrentPage(); // Highlight the current page in the navigation bar
```
Comment (melkati) [inline webserver/preferences.js:598]: Why is this commented out?

File: webserver/preferences.js [hunk 3]
```diff
@@ -619,3 +619,85 @@ document.addEventListener("DOMContentLoaded", () => {
...
     }
 });
+
+/**
+ * Sanity Data form before Save 
+ */
+function sanityData() {
+    var txt_error = "";
+    let errorMessage = document.getElementById('error-message');
+    const displayB = document.getElementById("DisplayBright");
```
Comment (melkati) [inline webserver/preferences.js:629]: Please avoid using cryptic names for variables such as "displayB". Variable names should be self-explanatory.

File: webserver/preferences.js [hunk 4]
```diff
@@ -619,3 +619,85 @@ document.addEventListener("DOMContentLoaded", () => {
...
+    if (parseInt(displayB.value) > displayB.max || parseInt(displayB.value) < displayB.min) {
+        displayB.classList.remove('valid');
+        displayB.classList.add('error');
```
Comment (melkati) [inline webserver/preferences.js:632]: Cryptic names

File: webserver/preferences.js [hunk 5]
```diff
@@ -619,3 +619,85 @@ document.addEventListener("DOMContentLoaded", () => {
...
+/**
+ * Runtime display brightness 
+ */
+function setDisplayBrightness() {
+    const displayB = document.getElementById("DisplayBright").value;
+    console.log("Set Display Brightness = " + displayB);
+    fetch(`/settings?setDisplayBrightness=${displayB}`)
+        .then(response => {
+            if (!response.ok) throw new Error('Error setting display brightness');
+            console.log('Set Display brightness successfully');
```
Comment (melkati) [inline webserver/preferences.js:700]: To avoid excessive and redundant logging, use the following condition:
if (preferencesDebug) console.log('Display brightness set successfully.');

File: webserver/style.css
Comment (melkati) [inline webserver/style.css:1]: You can use Prettier to format CSS code and maintain a consistent style.
https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

File: webserver/style.css [hunk 6]
```diff
@@ -678,3 +687,29 @@ button:hover {
 .hidden {
   display: none;
 }
+
+input.valid {
+  border-color: #468847;
```
Comment (melkati) [inline webserver/style.css:692]: Utilize CSS variables to ensure consistency and to maintain various color schemes for theming.
https://github.com/melkati/CO2-Gadget/blob/development/webserver/style.css#L1-L65

**Review by melkati — COMMENTED** [review]
Great work! Just a few minor changes needed. Thank you!

File: webserver/preferences.js [hunk 7]
```diff
@@ -619,3 +619,85 @@ document.addEventListener("DOMContentLoaded", () => {
...
+        }
+        return false;
+    } else {
+        if (errorMessage) errorMessage.remove();
+        displayB.classList.remove('error');
+        displayB.classList.add('valid');
+    }
+
+    const co2O = document.getElementById("co2OrangeRange");
+    const co2R = document.getElementById("co2RedRange");
```
Comment (melkati) [inline webserver/preferences.js:653]: Please avoid using cryptic names for variables such as co2O or co2R. Variable names should be self-explanatory.

File: webserver/preferences.js (hunk 2 above)
Comment (Coscolin) [reply webserver/preferences.js:598]: highlightCurrentPage() function is not defined in this scope and generates a compile error.

File: CO2_Gadget_WIFI.h (hunk 1 above)
Comment (Coscolin) [reply CO2_Gadget_WIFI.h:954]: Of course...
I didn't know you put forwards decalrations there.

File: webserver/preferences.js (hunk 2 above)
Comment (melkati) [reply webserver/preferences.js:598]: I assume you're referring to an error message in the browser console, not a compilation error. I haven't noticed it, but I will check after the merge (the highlightCurrentPage() function is defined in common.js, which is included with all JavaScript files).

File: CO2_Gadget_TFT.h [hunk 8]
```diff
@@ -655,6 +655,7 @@ void displayShowValues(bool forceRedraw = false) {
     if (forceRedraw) {
         // Serial.println("-->[TFT ] Displaying values. Force Redraw: " + String(forceRedraw ? "true" : "false"));
         tft.fillScreen(TFT_BLACK);  // Remove previous remains in the screen
+        spr.fillSprite(TFT_BLACK);  // Remove previous remains in the screen
```
Comment (melkati) [inline CO2_Gadget_TFT.h:658]: I'm unsure why this is included here since no sprites are created at this point. Could this commit be an error? It seems unrelated to this PR regardless.

**Review by melkati — APPROVED** [review]
There are still some changes I don't completely understand, but I will merge them into a new branch ([development-sanity-checks-web](https://github.com/melkati/CO2-Gadget/tree/development-sanity-checks-web)) before merging into development.

File: webserver/preferences.js (hunk 2 above)
Comment (Coscolin) [reply webserver/preferences.js:598]: From development branch I have received a JavaScript error, before that function is not defined in ever .js file in development branch.

File: CO2_Gadget_TFT.h (hunk 8 above)
Comment (Coscolin) [reply CO2_Gadget_TFT.h:658]: In S3 version, display is not updated when temp/hum/battery is hidden until menu in and out and display refreshed completely. 
Showing a hidden value is updated well and quickly. 
This line is a try to repair this minor bug.
